src/inventorymanager.cpp

```diff
@@ -531,6 +531,17 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 		}
 	}
 
+	if (from_inv.type == InventoryLocation::PLAYER || to_inv.type == InventoryLocation::PLAYER){
```

Comment:
This needs to be split into two ifs, checking `to_i == player->getWieldIndex()` when to_inv isn't the player inventory causes bugs.